### ISSUE TITLE
Canvas height problem with angular material

### ISSUE THREAD

**Issue by nvme978** [jtblin/angular-chart.js#169]
## Modules in use:
1. Angular Material 0.10.0
2. chart.js 1.0.2
3. angular-chart.js 0.7.2

## Requirement:
Trying to create a responsive chart for all screen sizes using angular material. 

## Problem:
The height of the <code>.chart-container</code> div and the <code>canvas</code> is not taking it's parent <code>div</code> height. 
I want the chart to use up the entire height that is available to it in the page. 

I cannot use absolute values for <code>height</code> because I want the chart to fit nicely on all screen sizes. Therefore, I'm using percentages.

## Angular Material html markup:
```
<div layout="column" layout-fill style="position:absolute">
	<div layout="column" layout-margin layout-align="end" flex>
		<canvas id="bar" class="chart chart-bar" data="chart.data" series="chart.series" labels="chart.labels" options="chart.options" legend="true" height="100%"></canvas>
	</div>
</div>
```

## Chart options:
```javascript
$scope.chart.options = {
        responsive: true,
        maintainAspectRatio: true,
        barDatasetSpacing: 1,
        barShowStroke: true,
        barStrokeWidth : 2,
        barValueSpacing : 5
    };
```

## Notes:
1.<code>maintainAspectRatio</code> has to be set to true because I'm facing the issue of chart-resize every time the data is changed. There are issues for this logged on the ChartJS git but none of the solutions have worked for me so far.
2. <code>responsive: true</code> because I want the chart to be responsive. 
3. No CSS overrides have been applied to the <code>div</code> or the <code>canvas</code>.

## Screenshot of the parent <code>div</code> element

![screen shot 2015-07-30 at 9 00 36 pm](https://cloud.githubusercontent.com/assets/13456525/8987316/1bfc5f26-36fe-11e5-8d2e-2aeaa82cce26.png)

## Screenshot of the <code>.chart-container</code> div

![screen shot 2015-07-30 at 7 34 00 pm](https://cloud.githubusercontent.com/assets/13456525/8987349/4af86a22-36fe-11e5-9e65-efd3514ed70b.png)

## Screenshot of the <code>canvas</code>

![screen shot 2015-07-30 at 7 36 57 pm](https://cloud.githubusercontent.com/assets/13456525/8987361/53a61692-36fe-11e5-9e27-82f8a2cbaa24.png)

## Screenshot on larger screen (which is OK, kinda)

![screen shot 2015-07-30 at 7 38 34 pm](https://cloud.githubusercontent.com/assets/13456525/8984982/9ad7668a-36f2-11e5-8878-7d01a6a25111.png)

# TL;DR

The chart should fill the <code>height</code> of the parent <code>div</code>.

**Comment by mzbyszynski** [comment]
I have some bar charts inside an `md-grid-list` (`md-grid-tile` really). I was able to get the height to fill the tile by adding `height="100%" width="100%"` to the canvas (you already have the `height` set, so that may be enough, and I also added this to our css file:

```css
md-grid-tile .chart-container {
    width: 90%;
    height: 90%;
}
```

not quite the same layout as yours, but you may want to try styling that `chart-container` class. 

If you put together a plunkr or something that shows the issue, I'd be happy to take a look as well.

Hope that helps!

**Comment by muldon** [comment]
mzbyszynski this solution solved my problem. Thanks !

**Comment by nvme978** [comment]
Thanks for the idea mzbyszynski. I ended up using the flexbox css and have the following styles for the <code>.chart-container</code>

```
.chart-container {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    min-width:80%;
}
```

But now I see that the canvas is stretched to fill the <code>.chart-container</code>.

![image](https://cloud.githubusercontent.com/assets/13456525/9001659/bb3f119a-3777-11e5-9971-c88be3d96f54.png)

**Comment by muldon** [comment]
I tried everything, there is really a bug concerning sizes of the charts. Charts don't respect sizes, no matter what. I thought I had solved with the @mzbyszynski solution but not... still waiting for a solution...

**Comment by jtblin** [comment]
Could you post a repro using the [jsbin template](http://jsbin.com/dufibi/3/edit?html,js,output)? I don't use "angular material" myself.

**Comment by muldon** [comment]
sorry I don't use jsbin template. Only angularjs + jquery.

**Comment by jtblin** [comment]
Well sorry but I can't debug issues without a repro case... Maybe you didn't understand what a repro case with jsbin is?

**Comment by muldon** [comment]
exactly, I really didn't understand what it is. My apologies.

**Comment by Quixomatic** [comment]
Did you try the layout-fill attribute that comes with angular material? it will add the layout-fill class to any element it is set on and take up the full space available to it.

**Comment by muldon** [comment]
No James,
I ended up changing the libraries versions and my problem was solved.
Thanks anyway...

**Comment by nmarshall23** [comment]
I have a similar problem. The chart does not fill the parent element. Here is a [jsbin](http://jsbin.com/zovamox/3/edit?html,css,output). In my example the parent div, with the class chartBox, has a black border around it. That white space around chart shouldn't be there.

**Comment by angular-ui-sortable-duke** [comment]
I got to the answer https://stylesheets.space/questions/the-elements-displayed-within-an-html-canvas-need-to-adapt-to-different-screen-sizes

**Comment by jtblin** [comment]
This issue was labeled 'repro needed' several years ago and no reproduction has been provided. Closing as stale.
